### PR TITLE
MAINT: Blacklist np.array_equal in tests

### PR DESCRIPTION
**Pull request by gfyoung**
Follow-up to #18087.

### CODE REVIEW

**Comment by gfyoung** [comment]
This initial commit should fail.  I want to check to make sure that the check works before fixing it.

File: ci/lint.sh [hunk 1]
```diff
@@ -72,7 +72,7 @@ if [ "$LINT" ]; then
     echo "Linting *.c and *.h DONE"
 
     echo "Check for invalid testing"
-    grep -r -E --include '*.py' --exclude testing.py '(numpy|np)\.testing' pandas
+    grep -r -E --include '*.py' --exclude testing.py '(numpy|np)\.testing|(numpy|np)\.array_equal' pandas/tests
```
Comment (jreback) [inline ci/lint.sh]: ``grep -r -E --include '*.py' --exclude testing.py '(numpy|np)(\.testing|\.array_equal)' pandas/tests/``

a little more consise, can you list in a comment the things we are looking, e.g

np.testing
np.array_equal

File: pandas/tests/internals/test_internals.py [hunk 2]
```diff
@@ -286,29 +286,6 @@ def test_delete(self):
         with pytest.raises(Exception):
             newb.delete(3)
 
-    def test_split_block_at(self):
-
-        # with dup column support this method was taken out
-        # GH3679
-        pytest.skip("skipping for now")
```
Comment (jreback) [inline pandas/tests/internals/test_internals.py]: no problem removing, though his should in theory work now

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/internals/test_internals.py (hunk 2 above)
Comment (gfyoung) [reply pandas/tests/internals/test_internals.py]: The test doesn't work period.  The function "split_block_at" doesn't exist anywhere.

File: ci/lint.sh (hunk 1 above)
Comment (gfyoung) [reply ci/lint.sh]: Done.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
lgtm. merge on green.

**Pull request merged**